agent/consul/internal_endpoint.go

```diff
@@ -101,7 +101,7 @@ func (m *Internal) NodeDump(args *structs.DCSpecificRequest,
 			for _, p := range listedPeerings {
 				index, importedDump, err := state.NodeDump(ws, &args.EnterpriseMeta, p.Name)
 				if err != nil {
-					return err
+					return fmt.Errorf("could not get a node dump for peer's %q nodes: %w", p.Name, err)
```

Comment:
Minor grammar point, the `'s` should be after the peer name:
`"could not get a node dump for peer %s's nodes: %w"`

But that aside we can shorten this:
`"could not get a node dump for peer %q: %w"`